service/history/historyEngine.go

```diff
@@ -527,7 +529,56 @@ func (e *historyEngineImpl) getMutableState(ctx context.Context,
 		StickyTaskListScheduleToStartTimeout: common.Int32Ptr(msBuilder.executionInfo.StickyScheduleToStartTimeout),
 	}
 
-	return result, nil
+	return
+}
+
+func (e *historyEngineImpl) InquireMutableState(ctx context.Context,
+	request *h.InquireMutableStateRequest) (retResp *h.InquireMutableStateResponse, retError error) {
+
+	domainID, err := validateDomainUUID(request.DomainUUID)
+	if err != nil {
+		return nil, err
+	}
+
+	execution := workflow.WorkflowExecution{
+		WorkflowId: request.Execution.WorkflowId,
+		RunId:      request.Execution.RunId,
+	}
+
+	cacheCtx, dbCtx, release, cacheHit, err := e.historyCache.getAndCreateWorkflowExecutionWithTimeout(ctx, domainID, execution)
+	if err != nil {
+		return nil, err
+	}
+	defer func() { release(retError) }()
+	retResp = &h.InquireMutableStateResponse{}
+
+	if cacheHit && cacheCtx.msBuilder != nil {
+		msb := cacheCtx.msBuilder
+		retResp.MutableStateInCache, retError = e.toMutableStateJSON(msb)
+		otherInfo := ""
+		if len(msb.updateActivityInfos) > 0 {
+			otherInfo += "updateActivityInfos is not empty;"
```

Comment:
why do we need this otherInfo? The toMutableStateJSON should already include everything, right?

Reply:
Like above. there are still someother information like updateActivityInfos. @wxing1292 you guys can discuss if they are really needed. I don't have too much context of how we are going to use it.

Reply:
well, i still prefer dumping everything as original type, not string